### PR TITLE
Errors

### LINKED ISSUE #75: Codici di errore
Come definiti in Avviso 41, da includere nella documentazione di Docs italia estendndo l'apposita sezione
https://spid-cie-oidc-docs.vercel.app/gestione_errori.html#error-codes

immagine

### PR DESCRIPTION
**Pull request by peppelinux**
Closes https://github.com/italia/spid-cie-oidc-docs/issues/75

this PR removes the Error macro section, specializing sub section for each endpoint.
It also enrich the definition, following SPID Avviso 41

- [x] Ensure your files are written following RST specs (*not MD!*)
- [x] Italian version
- [x] English version
- [x] Example files 
- [x] Ask for review

### CODE REVIEW

File: docs/en/authorization_endpoint.rst [hunk 1]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+In the event of an error, the OP or RP represent the anomaly message 
+as described in the related tables defined by the `Linee Guida UX SPID`_. 
+
+
+.. list-table:: 
+   :widths: 20 60 20
+   :header-rows: 1
+
+   * - **Claim**
+     - **Descrizione**
```
Comment (damikael) [inline docs/en/authorization_endpoint.rst]: Description

File: docs/en/authorization_endpoint.rst [hunk 2]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+     - **Supportato da**
```
Comment (damikael) [inline docs/en/authorization_endpoint.rst]: Supported by

File: docs/en/authorization_endpoint.rst [hunk 3]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+     - Error description.
+     - |spid-icon| |cieid-icon|
+   * - **state**
+     - It MUST be equal to the *status* value included in the *Authentication Request*. The RP MUST verify that it matches the one sent in the *Authentication Request*.
+     - |spid-icon| |cieid-icon|
+
+
+.. _codici_errore:
+
+Codici di errore
```
Comment (damikael) [inline docs/en/authorization_endpoint.rst]: Error codes

File: docs/en/authorization_endpoint.rst [hunk 4]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+----------------
+
+.. list-table:: 
+   :widths: 20 20 20 20
+   :header-rows: 1
+
+   * - **Error Code**
+     - **Descrizione**
```
Comment (damikael) [inline docs/en/authorization_endpoint.rst]: Description

File: docs/en/authorization_endpoint.rst [hunk 5]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+     - **Reference**
+     - **Supportato da**
```
Comment (damikael) [inline docs/en/authorization_endpoint.rst]: Supported by

File: docs/it/authorization_endpoint.rst [hunk 6]
```diff
@@ -276,3 +276,73 @@ Esempio di Authorization Response dell'OP:
...
+
+
+.. list-table:: 
+   :widths: 20 60 20
+   :header-rows: 1
+
+   * - **Claim**
+     - **Descrizione**
+     - **Supportato da**
+   * - **error**
```
Comment (damikael) [inline docs/it/authorization_endpoint.rst]: Errore

File: docs/it/authorization_endpoint.rst [hunk 7]
```diff
@@ -276,3 +276,73 @@ Esempio di Authorization Response dell'OP:
...
+     - Vedi :ref:`Codici di errori <codici_errore>`
+     - |spid-icon| |cieid-icon|
+   * - **error_description**
```
Comment (damikael) [inline docs/it/authorization_endpoint.rst]: Descrizione dell'errore

File: docs/it/authorization_endpoint.rst [hunk 8]
```diff
@@ -276,3 +276,73 @@ Esempio di Authorization Response dell'OP:
...
+.. _codici_errore:
+
+Codici di errore
+----------------
+
+.. list-table:: 
+   :widths: 20 20 20 20
+   :header-rows: 1
+
+   * - **Error Code**
```
Comment (damikael) [inline docs/it/authorization_endpoint.rst]: Codice errore

File: docs/it/authorization_endpoint.rst [hunk 9]
```diff
@@ -276,3 +276,73 @@ Esempio di Authorization Response dell'OP:
...
+     - **Descrizione**
+     - **Reference**
```
Comment (damikael) [inline docs/it/authorization_endpoint.rst]: Riferimento

**Review by damikael — COMMENTED** [review]
some translation

File: docs/it/authorization_endpoint.rst (hunk 9 above)
Comment (peppelinux) [reply docs/it/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/it/authorization_endpoint.rst (hunk 6 above)
Comment (peppelinux) [reply docs/it/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/en/authorization_endpoint.rst (hunk 5 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/en/authorization_endpoint.rst (hunk 4 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/en/authorization_endpoint.rst (hunk 3 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/en/authorization_endpoint.rst (hunk 2 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

File: docs/en/authorization_endpoint.rst (hunk 1 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst]: Fixed in https://github.com/italia/spid-cie-oidc-docs/pull/77/commits/19a3b3ea839f2763010e98c7706453b7b38d4036

**Review by damikael — APPROVED** [review]

**Comment by peppelinux** [comment]
Attendiamo la revisione di @fmarino-ipzs prima di unire

File: docs/en/authorization_endpoint.rst [hunk 10]
```diff
@@ -286,4 +284,71 @@ Authorization Response example:
...
+     - **Description**
+     - **Reference**
+     - **Supported by**
+
+   * - *access_denied*
+     - The OP denied access due to invalid or unsuitable credentials for the required SPID level.
+     - 
+     - |spid-icon| |cieid-icon|
+
+   * - *invalid_client*
```
Comment (fmarino-ipzs) [inline docs/en/authorization_endpoint.rst:331]: In OIDC Core questo codice non compare tra i codici errore previsti da standard. 
Nemmeno nella sezione 4.1.2.1 dell'RFC6749.
Proporrei:
_unauthorized_client
               The client is not authorized to request an authorization
               code using this method._

Aggiungerei inoltre:

_unsupported_response_type
               The authorization server does not support obtaining an
               authorization code using this method._

_login_required
The Authorization Server requires End-User authentication. This error MAY be returned when the prompt parameter value in the Authentication Request is none, but the Authentication Request cannot be completed without displaying a user interface for End-User authentication._

_consent_required
The Authorization Server requires End-User consent. This error MAY be returned when the prompt parameter value in the Authentication Request is none, but the Authentication Request cannot be completed without displaying a user interface for End-User consent._

_request_uri_not_supported
The OP does not support use of the request_uri parameter defined in [Section 6](https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests)._

_registration_not_supported
The OP does not support use of the registration parameter defined in [Section 7.2.1](https://openid.net/specs/openid-connect-core-1_0.html#RegistrationParameter)._

_invalid_request_uri
The request_uri in the Authorization Request returns an error or contains invalid data._

_invalid_request_object
The request parameter contains an invalid Request Object._

File: docs/en/token_endpoint.rst [hunk 11]
```diff
@@ -295,3 +295,39 @@ The claims available in the *ID Token* are given below.
...
+
+.. list-table:: 
+   :widths: 20 60 20
+   :header-rows: 1
+
+   * - **Claim**
+     - **Description**
+     - **Supported by**
+
+   * - *invalid_client*
```
Comment (fmarino-ipzs) [inline docs/en/token_endpoint.rst:311]: In OIDC Core questo codice fa riferimento alla Token Error Response
Ref:
RFC 6749 - Oauth 2.0 – 5.2
OIDC Core 1.0 – 3.1.3.4

File: docs/it/authorization_endpoint.rst [hunk 12]
```diff
@@ -276,3 +276,73 @@ Esempio di Authorization Response dell'OP:
...
+     - **Descrizione**
+     - **Riferimento**
+     - **Supportato da**
+
+   * - *access_denied*
+     - L’OP ha negato l’accesso a causa di credenziali non valide o non adeguate al livello SPID richiesto.
+     - 
+     - |spid-icon| |cieid-icon|
+
+   * - *invalid_client*
```
Comment (fmarino-ipzs) [inline docs/it/authorization_endpoint.rst:324]: Vedi commento nella versione inglese

File: docs/en/token_endpoint.rst [hunk 13]
```diff
@@ -295,3 +295,39 @@ The claims available in the *ID Token* are given below.
  - https://openid.net/specs/openid-igov-openid-connect-1_0-03.html#rfc.section.3.1
 
 
+.. _TOKEN_ENDPOINT_ERRORS:
```
Comment (fmarino-ipzs) [inline docs/en/token_endpoint.rst:298]: In base a https://www.rfc-editor.org/rfc/rfc6749#section-5.2 aggiungerei:

- unauthorized_client
               The authenticated client is not authorized to use this
               authorization grant type.

-

File: docs/it/token_endpoint.rst [hunk 14]
```diff
@@ -294,3 +294,40 @@ Di seguito i claim disponibili nell'ID Token.
  - https://openid.net/specs/openid-igov-openid-connect-1_0-03.html#rfc.section.3.1
 
 
+.. _TOKEN_ENDPOINT_ERRORS:
```
Comment (fmarino-ipzs) [inline docs/it/token_endpoint.rst:297]: Vedi commento su versione inglese

**Review by fmarino-ipzs — CHANGES_REQUESTED** [review]
In generale non so se sia meglio avere codici di errori specifici per tipologia di endpoint (token, userinfo, introspection, revocation). Andrebbero anche aggiunti i codici di errori relativi al logout.

File: docs/en/authorization_endpoint.rst (hunk 10 above)
Comment (peppelinux) [reply docs/en/authorization_endpoint.rst:331]: invalid_client è in OAuth2 
https://www.rfc-editor.org/rfc/rfc6749#section-5.2

unisco questa PR e converto il tuo commento in una issue integrativa per gli errori, così dividiamo il commti di avviso 41 con l'estensione da te proposta

**Pull request merged**